src/main/java/com/cms/claimmanagement/insuranceCompany/Claim/Controller/ClaimController.java

```diff
@@ -2,26 +2,34 @@
 
 import com.cms.claimmanagement.insuranceCompany.Claim.Service.ClaimService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
-@RequestMapping("/claims")
+@RequestMapping("/v1/claims")
 public class ClaimController {
     @Autowired
     ClaimService claimService;
 
     @GetMapping
-    public List<ClaimResponseData> getAllClaims(){
-       return claimService.getAllClaims();
+    public ResponseEntity<List<ClaimResponseData>> getOpenClaims(){
+        List<ClaimResponseData> listOfOpenClaims = claimService.getAllOpenClaims();
+       return ResponseEntity.ok().body(listOfOpenClaims);
 
     }
 
     @PostMapping(path = "/new")
-    public ClaimResponseData createNewClaim(@RequestBody ClaimRequestData claimRequest){
+    public ResponseEntity<ClaimResponseData> createNewClaim(@RequestBody ClaimRequestData claimRequest){
+         ClaimResponseData savedClaim = claimService.saveNewClaim(claimRequest);
+         return ResponseEntity.ok().body(savedClaim);
+    }
+    @PutMapping(path = "/{claimId}/update")
+    public ResponseEntity<ClaimUpdateData> updateClaim(@PathVariable("claimId") String claimId, @RequestBody ClaimUpdateData updateData){
```

Comment:
Why are you receving a claimId if it's not being used?

Reply:
true